magefile.go

```diff
@@ -486,8 +486,15 @@ func (Test) Unit(ctx context.Context) error {
 
 // FIPSOnlyUnit runs all the unit tests with GODEBUG=fips140=only.
 func (Test) FIPSOnlyUnit(ctx context.Context) error {
+	// We pre-cache go module dependencies before running the unit tests with
+	// GODEBUG=fips140=only.  Otherwise, the command that runs the unit tests
+	// will try to download the dependencies and could fail because the TLS
+	// negotiation with the Go module proxy could use a non-FIPS compliant
+	// key exchange protocol, e.g. X25519.
+	RunGo("mod", "download")
```

Comment:
Nit: as an alternative, the test binary could be built without the `GODEBUG` and `FIPS` env vars prior to running th actual `go test` command

Reply:
True. I went with this approach because I can keep it consistent with Fleet Server and Beats where I don't believe we're explicitly building test binaries for unit tests.